core/src/main/java/org/apache/iceberg/UpdateSnapshotReferencesOperation.java

```diff
@@ -106,40 +106,41 @@ public UpdateSnapshotReferencesOperation replaceBranch(String name, long snapsho
     return this;
   }
 
-  public UpdateSnapshotReferencesOperation replaceBranch(String name, String source) {
-    return replaceBranch(name, source, false);
+  public UpdateSnapshotReferencesOperation replaceBranch(String from, String to) {
+    return replaceBranch(from, to, false);
   }
 
-  public UpdateSnapshotReferencesOperation fastForward(String name, String source) {
-    return replaceBranch(name, source, true);
+  public UpdateSnapshotReferencesOperation fastForward(String from, String to) {
+    return replaceBranch(from, to, true);
   }
 
   private UpdateSnapshotReferencesOperation replaceBranch(
-      String name, String source, boolean fastForward) {
-    Preconditions.checkNotNull(name, "Target branch cannot be null");
-    Preconditions.checkNotNull(source, "Source ref cannot be null");
-    SnapshotRef sourceRef = updatedRefs.get(source);
-    SnapshotRef refToUpdate = updatedRefs.get(name);
-    Preconditions.checkArgument(refToUpdate != null, "Target branch does not exist: %s", name);
-    Preconditions.checkArgument(sourceRef != null, "Ref does not exist: %s", source);
-    Preconditions.checkArgument(refToUpdate.isBranch(), "Ref %s is a tag not a branch", name);
+      String from, String to, boolean fastForward) {
+    Preconditions.checkNotNull(from, "Branch to update cannot be null");
```

Comment:
I think we're a little inconsistent here in terms of null checking (throwing NPE vs IAE) but I guess that's out-of-scope for this PR, so just wanted to mention that

Reply:
Yeah you're right, and this unfortunately applies to more than just replace/fastForward. (e.g. create, rename, as well throw an NPE instead of IAE).  I'll take this in a follow on.